pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/ClassWithOnlyPrivateConstructorsShouldBeFinalRule.java

```diff
@@ -21,9 +23,10 @@ public ClassWithOnlyPrivateConstructorsShouldBeFinalRule() {
     @Override
     public Object visit(ASTClassOrInterfaceDeclaration node, Object data) {
         if (node.isRegularClass()
-            && !node.isFinal()
+            && !node.hasModifiers(JModifier.FINAL)
             && hasOnlyPrivateCtors(node)
-            && hasNoSubclasses(node)) {
+            && hasNoSubclasses(node)
+            && hasNoAbstractMethods(node)) {
```

Comment:
Why not `node.isAbstract()`? A class has to be abstract to contain abstract methods